app/src/main/java/pl/targosz/invoicing/database/file/FileBasedInvoiceRepository.java

```diff
@@ -0,0 +1,91 @@
+package pl.targosz.invoicing.database.file;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import pl.targosz.invoicing.config.FileConfiguration;
+import pl.targosz.invoicing.database.InvoiceRepository;
+import pl.targosz.invoicing.model.Invoice;
+import pl.targosz.invoicing.services.FileService;
+import pl.targosz.invoicing.services.JsonService;
+
+@AllArgsConstructor
+public class FileBasedInvoiceRepository implements InvoiceRepository {
+
+    private final FileService invoicesFileService = new FileService(FileConfiguration.INVOICES_DB_PATH);
```

Comment:
Maybe use @RequiredArgsConstructor and initialize FileServices and JsonService when creating FileBasedInvoiceRepository class instance

Reply:
Done